src/scheduler.rs

```diff
@@ -590,6 +617,9 @@ impl Scheduler {
                 if let Some(handle) = smtp_handle.take() {
                     handle.await;
                 }
+                if let Some(handle) = ephemeral_handle.take() {
+                    handle.cancel().await;
```

Comment:
I don't think this will ever finish, because you are not stopping the loop inside the ephemeral_handle

Reply:
there is a chance it works out at the moment, but definitely not a clean shutdown

Reply:
My understanding is that `cancel()` drops the future, which means that at the next suspend point - in this case mainly during the `timeout(...).await` or the `interrupt_receiver.recv().await`, the task is stopped. Is this correct?

Reply:
I believe so, but I am not sure if this `await` actually waits for that to happen

Reply:
From the documentation:

> Cancels the task and waits for it to stop running.
>     
> Returns the task's output if it was completed just before it got canceled, or [`None`] if
> it didn't complete.
>     
> While it's possible to simply drop the [`Task`] to cancel it, this is a cleaner way of
> canceling because it also waits for the task to stop running.

<details>

`JoinHandle::cancel()` has no documentation, but calls through to `async_task::task::Task::cancel()` which is documented here: https://docs.rs/async-task/latest/async_task/struct.Task.html#method.cancel

</details>

Reply:
so i think this is resolved and we can leave it as is.  The functional python tests also pass (exercising stop_io) so it seems to be ok.

Reply:
Also note that we called `cancel()` in `stop_io()` before: https://github.com/deltachat/deltachat-core-rust/blob/35c0434dc79441547a2580a590af0f781fb204b9/src/context.rs#L646-L648

(this was for the old ephemeral task logic)

Reply:
Stopping ephemeral loop can be done safer indeed, `cancel()` interrupts the loop wherever it is, maybe even in the middle of executing SQL statements, basically at any `.await` even deep inside the call stack. Telling the loop to stop in `pre_stop` is a better solution.